.travis.yml

```diff
@@ -10,25 +10,25 @@ jobs:
     - name: "Build, Test, and Run Coverage"
       script:
         - cd src
-        # Build, test, run coverage
+        # Test Software
         - >
-          bazel coverage //software/...
-          --compilation_mode=fastbuild 
-          --verbose_test_summary 
-          --instrumentation_filter=//... 
-          --test_output=errors
-          --experimental_use_llvm_covmap
-          --collect_code_coverage
-        # Convert profdata results to lcov. We have to do this because of the
-        # current state of support for clang compiler coverage results in
-        # bazel. Once https://github.com/bazelbuild/bazel/issues/9406 is 
-        # addressed, we can probably change the `coverage_feature` flags in
-        # `cc_toolchain_config.bzl` to just `--coverage` and remove this script
-        - ./convert_profdata_to_lcov.sh
-        # Upload coverage results
-        - bash <(curl -s https://codecov.io/bash)
-        # Build our CPU-specific targets
-        - bazel build //firmware_new/... --cpu=stm32h7
+          bazel coverage //software/...     \
```

Comment:
just want to double check. Bazel's CLI-Reference says the `coverage` command "Generates code coverage report for specified test targets". It seems it's roughly the same as `test` but with coverage. Does this mean we're not building _all_ the code, ie. missing the top-level code not associated with tests?

I know we need to build code with covarege enabled to get the coverage data (so doesn't make sense to necessarily call `build` here).

Reply:
Correct, I just thought  of this while I was cleaning this up.......... 

We _could_ run ` bazel build //software:full_system_main` as well, but the different command line options would probably trigger a re-compile of basically everything.

Reply:
Maybe worth trying locally. Or perhaps we can somehow pass the coverage flags to `build` first? My understanding is that the main difference is the extra symbols / tracking built in for coverage to consume

Reply:
Just made some changes to `full_system_main.cpp` and it failed. No test depends on that target, so looks like bazel builds everything. So this is actually a non-issue! :tada: